libglusterfs, rpc: drop old string data format

dmantipov opened the pull request:
Drop `GF_DATA_TYPE_STR_OLD` dict data type and related code,
prefer `GF_DATA_TYPE_STR` in volgen and serialization.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
Are there real reasons for keeping `GF_DATA_TYPE_STR_OLD`? Any weird kind of backward compatibility?

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/rpc/xdr/src/glusterfs3.h b/rpc/xdr/src/glusterfs3.h
index 3a02aab17..1c0e843c9 100644
--- a/rpc/xdr/src/glusterfs3.h
+++ b/rpc/xdr/src/glusterfs3.h
@@ -52,7 +52,6 @@
 /* refer 2717 */
 #define GF_O_PATH 010000000
 
-
 #define XLATE_BIT(from, to, bit)                                               \
     do {                                                                       \
         if (from & bit)                                                        \
```

dmantipov commented:
/run regression

dmantipov commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/dict.c b/libglusterfs/src/dict.c
index 1afadc33b..d5dca63d8 100644
--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -2875,7 +2875,6 @@ out:
  *          failure: -errno
  */
 
-
 /**
  * dict_unserialize - unserialize a buffer into a dict
  *
```

gluster-ant commented:
1 test(s) failed 
./tests/bugs/core/bug-1650403.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/basic/afr/ta-shd.t
./tests/bugs/core/bug-1650403.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2471/

dmantipov commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/dict.c b/libglusterfs/src/dict.c
index 1afadc33b..d5dca63d8 100644
--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -2875,7 +2875,6 @@ out:
  *          failure: -errno
  */
 
-
 /**
  * dict_unserialize - unserialize a buffer into a dict
  *
```

xhernandez reviewed (dismissed):
If I'm not wrong, this change will break network protocol, so upgrading clients and servers at different times will probably cause issues.

amarts commented on libglusterfs/src/glusterfs/glusterfs-fops.h:
```diff
@@ -225,16 +225,15 @@ typedef enum gf_upcall_flags_t gf_upcall_flags_t;
+    GF_DATA_TYPE_GFUUID = 6,
+    GF_DATA_TYPE_IATT = 7,
+    GF_DATA_TYPE_MDATA = 8,
+    GF_DATA_TYPE_MAX = 9,
```
This may have an impact, as its part of protocol, and server (or) client would be still using old version. Can we retain the enum same with that variable removed, but values of other variables remain same?

dmantipov replied:
Sure.

Commit:
libglusterfs, rpc: drop old string data format

Drop GF_DATA_TYPE_STR_OLD dict data type and related code,
prefer GF_DATA_TYPE_STR in volgen and serialization.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/dict.c b/libglusterfs/src/dict.c
index 057d38a23..2c0d48533 100644
--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -2835,8 +2835,6 @@ out:
     return ret;
 }
 
-
-
 /**
  * dict_unserialize - unserialize a buffer into a dict
  *
diff --git a/libglusterfs/src/glusterfs/dict.h b/libglusterfs/src/glusterfs/dict.h
index 12463b2d6..2365fa9da 100644
--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -173,7 +173,6 @@ dict_reset(dict_t *dict);
 int
 dict_key_count(dict_t *this);
 
-
 int32_t
 dict_unserialize(char *buf, int32_t size, dict_t **fill);
 
```

xhernandez commented on libglusterfs/src/glusterfs/glusterfs-fops.h:
```diff
@@ -225,7 +225,8 @@ typedef enum gf_upcall_flags_t gf_upcall_flags_t;
     GF_DATA_TYPE_UNKNOWN = 0,
-    GF_DATA_TYPE_STR_OLD = 1,
+    /* Backward compatibility placeholder, do not use. */
+    GF_DATA_TYPE_STR_OLD = -1,
```
I think the value shouldn't change, just to keep documented what was the old value. It won't be used anyway.

xhernandez reviewed (commented):
I'm still not sure if there can't be any place where the old `GF_DATA_TYPE_STR_OLD` value may be sent through the network or stored somewhere. In that case we should keep recognizing it to avoid upgrade issues.

However I'm not familiar enough with this part of the code and I don't know the reach of that dict field.

I'm removing my -1 to not block this patch if it's actually fine.
